scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamWriter.scala

```diff
@@ -0,0 +1,60 @@
+package com.twitter.scalding.beam_backend
+
+import com.stripe.dagon.Rule
+import com.twitter.scalding.Execution.{ ToWrite, Writer }
+import com.twitter.scalding.typed._
+import com.twitter.scalding.{ CFuture, CancellationHandler, Config, Execution, ExecutionCounters }
+import java.util.concurrent.atomic.AtomicLong
+import org.apache.beam.sdk.Pipeline
+import scala.concurrent.{ ExecutionContext, Future }
+
+class BeamWriter(val beamMode: BeamMode) extends Writer {
+  private val state = new AtomicLong()
+
+  override def start(): Unit = ()
+
+  override def finished(): Unit = ()
+
+  def getForced[T](conf: Config, initial: TypedPipe[T])(implicit cec: ExecutionContext): Future[TypedPipe[T]] = ???
+
+  def getIterable[T](conf: Config, initial: TypedPipe[T])(implicit cec: ExecutionContext): Future[Iterable[T]] = ???
+
+  override def execute(
+    conf: Config,
+    writes: List[ToWrite[_]])(implicit cec: ExecutionContext): CFuture[(Long, ExecutionCounters)] = {
+    import Execution.ToWrite._
+    val planner = BeamPlanner.plan(conf, beamMode.sources)
+    val phases: Seq[Rule[TypedPipe]] = BeamPlanner.defaultOptimizationRules(conf)
+    val optimizedWrites = ToWrite.optimizeWriteBatch(writes, phases)
+    val pipeline = Pipeline.create(beamMode.pipelineOptions)
+
+    def rec(optimizedWrites: List[OptimizedWrite[TypedPipe, _]]): Unit = {
+      optimizedWrites match {
+        case Nil => Unit
```

Comment:
`Unit` is the type and companion object for the type, `()` is the value. This is working because scala is casting the companion object into the value unit.

It is better to write: `case Nil => ()` here.

that is not a shortcut, is actually the correct thing that works without relying on Any to Unit casting.

It's a bit subtle, but the key is the distinction between the type and companion object name `Unit` and the value `()`. Honestly, I think scala messed this up, and I think it would have been better for the unit value to be called `Unit`... but oh well.

that would have been something like this:
```scala
// here the type is defined
sealed trait Unit

// here is a value, note the name space of values and types are different, so they can somewhat confusingly have
// the same name.
object Unit extends Unit
```